Moved from v1alpha1 to v1

jpkrohling opened the pull request:
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

jpkrohling commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/jaegertracing/jaeger-operator/265)
<!-- Reviewable:end -->

jpkrohling commented:
This PR is still missing the warning in case people try to use the old `v1alpha1`. This will be done in a subsequent commit to this PR, but it would be nice to get a first feedback on the changes so far.

jpkrohling commented:
When a CR with the old `apiVersion` is deployed, this is what's shown in the operator's logs:

```
WARN[0000] Attention: you are using a legacy CR for Jaeger. Use the apiVersion 'jaegertracing.io/v1' instead of 'io.jaegertracing/v1alpha1'  instance=simplest namespace=default
```

objectiser commented on test/e2e/sidecar.go:
```diff
@@ -161,6 +161,7 @@ func sidecarTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) e
 		resp := &resp{}
 		err = json.Unmarshal(body, &resp)
 		if err != nil {
+			fmt.Println(body)
```
Did you mean to have this `Println`?

jpkrohling commented:
> Just want to clarify - so the current PR will print a warning and not deploy anything, but before we release v1 it will handle deploying based on the old alpha1 version aswell?

We *could* do some type transformation. My preference would be to *not* have it for `v1alpha1` -> `v1`, only for `v1` -> `v1.next` and on.

jpkrohling commented on CONTRIBUTING.adoc:
```diff
@@ -90,7 +90,7 @@ Similar instructions also work for OpenShift, but the target `run-openshift` can
 ==== Model changes
 
-The Operator SDK generates the `pkg/apis/io/v1alpha1/zz_generated.deepcopy.go` file via the command `make generate`. This should be executed whenever there's a model change (`pkg/apis/io/v1alpha1/types.go`)
+The Operator SDK generates the `pkg/apis/jaegertracing/v1/zz_generated.deepcopy.go` file via the command `make generate`. This should be executed whenever there's a model change (`pkg/apis/jaegertracing/v1/jaeger_types.go`)
```
Once we migrate to the SDK 0.5.0, I'll change this, as there are more generated files.

jpkrohling commented on deploy/crds/jaegertracing_v1_jaeger_cr.yaml:
```diff
@@ -0,0 +1,4 @@
+apiVersion: jaegertracing.io/v1
```
That's where people expect the example to be. The "scorecard" also checks for a CR here.

jpkrohling commented on deploy/crds/jaegertracing_v1_jaeger_crd.yaml:
```diff
@@ -0,0 +1,33 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Users would have the CRD already, but if we need it for some test (like, when checking if we have a warning message if an old CR is installed), then we need this CRD installed in the cluster.

jpkrohling commented on deploy/crds/jaegertracing_v1_jaeger_crd.yaml:
```diff
@@ -0,0 +1,33 @@
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
```
I'll remove it, this comes from the SDK 0.5.0, which I used as reference for the new API.

jpkrohling commented:
> Do we need to keep apis.go for v1aplha1 if we don't do model transhormation?

Yes, otherwise users will get the following error when trying to create a Jaeger instance with the old `apiGroup`:

```
error: unable to recognize "/tmp/simplest.yaml": no matches for kind "Jaeger" in version "io.jaegertracing/v1"
```

> It seems that some files reference io.jaegertracing. Like the OLM catalog

I fixed the OLM catalog, which will still include the old `apiGroup` for the time being. I fixed the one in the contribution guide as well. All other entries look fine to me. Could you please re-check?

pavolloffay commented:
> I fixed the OLM catalog, which will still include the old apiGroup for the time being. 

We should maybe create an issue to migrate it to the new one?

jpkrohling commented:
> We should maybe create an issue to migrate it to the new one?

I meant, it has both `apiGroups` now.
